services/QuillLMS/client/app/bundles/Staff/components/evidence/rulesAnalysis/rulesAnalysis.tsx

```diff
@@ -104,6 +117,12 @@ const RulesAnalysis: React.FC<RouteComponentProps<ActivityRouteProps>> = ({ hist
     queryFn: fetchActivity
   });
 
+  React.useEffect(() => {
+    onStartDateChange(new Date(getDateFromLatestAutoMLModel(modelsData?.models)))
+    setStartDate(getDateFromLatestAutoMLModel(modelsData?.models))
```

Comment:
We could clean up this code a bit by collapsing the `startDate` and `startDateForQuery` state hooks into a single hook. I *think* the only reason why we have both is that one is string format, and the other is date format, which we could address by adding a conversion function between the two (since downstream components expect both different formats).

Reply:
These were originally two separate pieces of state because we didn't want to trigger a request on first load until after the filter button was clicked. The date could be changed multiple times in the datepicker, stored to `startDate` and then `startDateForQuery` is set once the user clicks filter, triggering the request and bypassing the early return [here ](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/client/app/bundles/Staff/utils/evidence/ruleFeedbackHistoryAPIs.ts#L6)

If that's no longer the case and we want to trigger this request on load, then you should be fine to collapse them into one.

Reply:
These were originally two separate pieces of state because we didn't want to trigger a request on first load until after the filter button was clicked. The date could be changed multiple times in the datepicker, stored to `startDate` and then `startDateForQuery` is set once the user clicks filter, triggering the request and bypassing the early return [here ](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/client/app/bundles/Staff/utils/evidence/ruleFeedbackHistoryAPIs.ts#L6)

If that's no longer the case and we want to trigger this request on load, then you should be fine to collapse them into one.

Reply:
also I believe you can just wrap these two lines in an if statement checking for the existence of `modelsData`, and if you pull the `getDateFromLatestAutoMLModel` function inside `RulesAnalysis`, you won't need to pass these arguments since you'll have access to `modelsData` within:

```
  React.useEffect(() => {
    if(modelsData && modelsData.models) {
      onStartDateChange(new Date(getDateFromLatestAutoMLModel()))
      setStartDate(getDateFromLatestAutoMLModel())
    }

  }, [modelsData])

  function getDateFromLatestAutoMLModel() {
    const { models } = modelsData
    const latestModel = models.find(model => model.state === 'active')
    if (!latestModel) { return null }
    return latestModel.created_at
  }
  ```

Reply:
I've added the if statement check - good idea. I'm keeping `getDateFromLatestAutoMLModel` outside of the RulesAnalysis component, so that it can remain a pure function.